externs/browser/window.js

```diff
@@ -181,22 +181,32 @@ function prompt(message, opt_value) {}
 function setImmediate(callback, callbackParams) {}
 
 /**
- * @param {Function|!TrustedScript|string} callback
+ * @typedef {function(...*):undefined}
+ */
+var TimerCallbackFunction;
```

Comment:
This is just a style nit, but I don't find this particular typedef worth it if it's just used in the one place below.

Reply:
This is true but introducing this typedef generates much better code in Elemental2. With a typedef, Elemental2 will generate a java interface named `elemental2.dom.TimerCallbackFunction`. Without the typedef it generates multiple synthetic types based on the usage location. In this case it would generate `elemental2.dom.DomGlobal.SetIntervalCallbackFn` even when passing to `setTimeout` function. And if we reordered the definitions in the extern it would generate different code (i.e. `elemental2.dom.DomGlobal.SetTimeoutCallbackFn`). Having a stable name is a much better developer experience IMHO.

However if this needs to be change to get the PR accepted, I can do it.

Reply:
Could you add a comment to help the reader understand the point of this declaration?
I think it would look like this.

```javascript
/**
 * A function to invoke when a timer expires.
 *
 * NOTE: The typing is explicitly loose in order to allow a function with any signature (params and return type) to be
 * used. Having an explicit typedef for use in the declarations of the timer methods below instead of just using
 * Function there helps Elemental2 to do a better job synthesizing Java code for this interface.
 */
var TimerCallbackFunction;
```